react/ProductName.tsx

```diff
@@ -102,31 +113,60 @@ function ProductName({
     )
   }
 
+  const linkProps = {
+    page: 'store.product',
+    params: {
+      slug: productLink,
+      id: productId,
+    },
+  }
+
   return (
     <Wrapper
       className={`${handles.productNameContainer} mv0 ${className ?? ''}`}
     >
-      <span className={`${handles.productBrand} ${brandNameClass ?? ''}`}>
-        {name} {showBrandName && brandName && `- ${brandName}`}
-      </span>
-      {showSku && skuName && (
-        <span className={`${handles.productBrand} ${skuNameClass ?? ''}`}>
-          {skuName}
-        </span>
-      )}
-      {showProductReference && productReference && (
-        <span
-          className={`${handles.productReference} ${
-            productReferenceClass ?? ''
-          }`}
-        >
-          {`REF: ${productReference}`}
+      <LinkWrapper
+        displayMode={displayMode}
+        className={`${handles.productNameLink} pointer c-link hover-c-link active-c-link no-underline underline-hover`}
+        linkProps={linkProps}
+      >
+        <span className={`${handles.productBrand} ${brandNameClass ?? ''}`}>
+          {name} {showBrandName && brandName && `- ${brandName}`}
         </span>
-      )}
+        {showSku && skuName && (
+          <span className={`${handles.productBrand} ${skuNameClass ?? ''}`}>
+            {skuName}
+          </span>
+        )}
+        {showProductReference && productReference && (
+          <span
+            className={`${handles.productReference} ${
+              productReferenceClass ?? ''
+            }`}
+          >
+            {`REF: ${productReference}`}
+          </span>
+        )}
+      </LinkWrapper>
     </Wrapper>
   )
 }
 
+/**
+ * Shows the link associated with the product name or plain text depending on the displayMode.
+ */
+const LinkWrapper = ({ displayMode, linkProps, className, children }: any) => {
```

Comment:
Can you type LinkWrapper props instead of any? Would be nice for readability and typecheck

Reply:
Done! :)